src/pages/components/BookmarkEditDialog.qml

```diff
@@ -53,26 +67,21 @@ UserPrompt {
         TextField {
             id: urlField
             text: url
-            width: parent.width
 
-            //: Placeholder for textfield to edit bookmark's URL
-            //% "Enter URL"
-            placeholderText: qsTrId("sailfish_browser-ph-url_edit")
+            acceptableInput: text.length > 0
+            onActiveFocusChanged: if (!activeFocus) errorHighlight = !acceptableInput
+            onAcceptableInputChanged: if (acceptableInput) errorHighlight = false
 
             //: Label for textfield to edit bookmark's URL
             //% "URL"
             label: qsTrId("sailfish_browser-la-url_editor")
+            //% "URL is required"
+            description:  errorHighlight ? qsTrId("sailfish_browser-la-url_editor_error") : ""
             inputMethodHints: Qt.ImhNoAutoUppercase | Qt.ImhUrlCharactersOnly
 
-            EnterKey.iconSource: canAccept ? "image://theme/icon-m-enter-accept"
-                                           : "image://theme/icon-m-enter-next"
-            EnterKey.onClicked: {
-                if (canAccept) {
-                    accept()
-                } else {
-                    titleField.focus = true
-                }
-            }
+            EnterKey.enabled: canAccept
+            EnterKey.iconSource: "image://theme/icon-m-enter-accept"
+            EnterKey.onClicked: accept()
```

Comment:
An enter can still be coming from the hardware keyboard. Accept() doesn't seem to protect with canAccept -> all cases like this should be protected?

Reply:
Or wait, was there an acceptBlocked happening.

Reply:
Yeah calling accept() will do same thing as attempting to accept with the dialog header buttons, show error. We would actually need dimmed mode for accept enter key that allowed clicking so could show error highlight there as well.